project/Build.scala

```diff
@@ -104,7 +104,7 @@ object Build {
    *  set to 3.1.3. If it is going to be 3.1.0, it must be set to the latest
    *  3.0.x release.
    */
-  val previousDottyVersion = "3.4.0"
+  val previousDottyVersion = "3.4.1"
```

Comment:
This is what we have followed so far. However, it seems that this workflow could accidentally hide binary incompatibilities. I will open a separate PR proposing a change to make this more robust.

Reply:
See #20068